File: python/2.blockinfo.py
# bdkpython 0.26.0
#
# Query an Electrum server for the testnet blockchain height and latest block hash.

import bdkpython as bdk
from utilities import check_bdk_version

check_bdk_version(original_major_version=0, original_minor_version=26)

blockchain_config = bdk.BlockchainConfig.ELECTRUM(
    bdk.ElectrumConfig(
        url="ssl://electrum.blockstream.info:60002",
        socks5=None,
        retry=5,
        timeout=None,
        stop_gap=100,
        validate_domain=True,
    )
)
blockchain = bdk.Blockchain(blockchain_config)

block_height = blockchain.get_height()
block_hash = blockchain.get_block_hash(block_height)

print(f"Latest block height is {block_height}")
print(f"Latest block hash is {block_hash}")

File: swift/2.blockinfo.swift
#!/usr/bin/swift sh

import BitcoinDevKit  // https://github.com/bitcoindevkit/bdk-swift.git ~0.26.0

let db = DatabaseConfig.memory

do {
    let blockchainConfig = BlockchainConfig.electrum(
        config: ElectrumConfig(
            // url: "ssl://electrum.blockstream.info:50002",
            url: "ssl://electrum.blockstream.info:60002",
            socks5: nil,
            retry: 5,
            timeout: nil,
            stopGap: 10,
            validateDomain: true
        )
    )
    
    let blockchain = try Blockchain(config: blockchainConfig)
    let blockHeight: UInt32 = try blockchain.getHeight()
    let blockHash: String = try blockchain.getBlockHash(height: blockHeight)
    
    print("Latest block is block $blockHeight with hash \(blockHash)")
} catch {
    print(error)
}
